### ISSUE TITLE
FR: Suggest compact()

### ISSUE THREAD

**Issue by brandonkelly** [kalessil/phpinspectionsea#578]
It would be cool if the plugin could identify arrays that could be simplified with [compact()](http://php.net/manual/en/function.compact.php) and suggest using compact() instead (and provide a quick-fix to do the work for you).

For example if it found this array, where all of the keys are identically-named to properties that their values are set to:

```php
[
    'foo' => $foo,
    'bar' => $bar
];
```

then the plugin could suggest simplifying it to:

```php
compact('foo', 'bar');
```

Seems like this would fit right in with the other maintainability features of the plugin.

**Comment by kelunik** [comment]
I'm not sure whether this should be recommended. `compact()` doesn't warn on undefined variables and relies on magic.

**Comment by funivan** [comment]
@kelunik pass data to the template (view)

**Comment by kalessil** [comment]
@kelunik : and we do report non-exisitng/undefined variables in "'compact(...)' variables existence"

**Comment by kalessil** [comment]
@brandonkelly : will be in the next release =)

**Comment by brandonkelly** [comment]
Yay! Thanks so much, Vlad.

**Comment by kalessil** [comment]
Implemented

**Comment by Bob4ever** [comment]
How does this improve the maintainability in any thinkable way?

I think **compact()** does quite the opposite. You couple the variable name with the key in the array. In clean code your variables (and their names) are just inside your method or function. But with **compact()** the name of your variable influence other code parts because it is also the key of your array.

And just imagine someone doesn't like (or use) the variable rename functionality. A simple search for "$variableName" will not be enough (like in the usual **compact()**-less way).

It's a nice feature but it doesn't improve anything except a litte more cryptic code.

**Comment by funivan** [comment]
@Bob4ever different teams use different rules. I do not use compact at all. But for someone this technique is pretty nice. 
Don`t worry inspection disabled by default.

**Comment by Bob4ever** [comment]
Then something is broken in my configuration :-)

It was activated otherwise I would've just ignored it.

**Comment by funivan** [comment]
@Bob4ever Hm. I just checked configuration and you are right Inspection is enabled by default `enabledByDefault="true"`

@kalessil maybe we should disable it for all users?

**Comment by kalessil** [comment]
@funivan : done.
@Bob4ever : maintainability improvements related to the case when we dispatching template arguments.
